tools/release/prompt/new-version-prompt.ts

```diff
@@ -0,0 +1,96 @@
+import {parseVersionName, VersionInfo} from '../version-name/parse-version';
+import {bold, red, yellow} from 'chalk';
+import {ChoiceType, prompt} from 'inquirer';
+import {createNewVersion} from '../version-name/create-version';
+import {ReleaseType, validateExpectedVersion} from '../version-name/check-version';
+
+/** Answers that will be prompted for. */
+type VersionPromptAnswers = {
+  releaseType: ReleaseType;
+  versionName: string;
+};
+
+/** Available options for selecting a release type. */
+const releaseTypeChoices = {
+  custom: {value: 'custom', name: 'Release w/ custom version.'},
+  stable: {value: 'stable', name: `Stable release`},
+  major: {value: 'major', name: 'Major release'},
+  minor: {value: 'minor', name: 'Minor release'},
+  patch: {value: 'patch', name: 'Patch release'},
+};
+
+/**
+ * Prompts the current user-input interface for a new version name. The new version will be
+ * validated to be a proper increment of the specified current version.
+ */
+export async function promptForNewVersion(versionName: string): Promise<VersionInfo> {
+  const currentVersion = parseVersionName(versionName);
+  let allowedReleaseTypes: ChoiceType[] = [releaseTypeChoices.custom];
+
+  if (!currentVersion) {
+    console.warn(red(`Cannot parse current project version. This means that we cannot validate ` +
+      `the new ${bold('custom')} version that will be specified.`));
+  } else if (currentVersion.suffix) {
+    console.warn(yellow(`Since the current project version is a ` +
+      `"${bold(currentVersion.suffix)}", the new version can be either custom or just the ` +
+      `stable version.`));
+    allowedReleaseTypes.unshift(releaseTypeChoices.stable);
+  } else {
+    allowedReleaseTypes.unshift(
+      releaseTypeChoices.major, releaseTypeChoices.minor, releaseTypeChoices.patch);
+  }
+
+  const answers = await prompt<VersionPromptAnswers>([{
+    type: 'list',
+    name: 'releaseType',
+    message: `What's the type of the new release?`,
+    choices: allowedReleaseTypes,
```

Comment:
That's a great idea. Currently I always prompt for the version name _again_. Since the version is already definitive if the `Major` option has been selected, we can avoid this and also show which version will be used if `Major` is selected.